dali/kernels/transpose/transpose_gpu_test.cc

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/kernels/transpose/transpose_gpu.h"  // NOLINT
+#include <gtest/gtest.h>
+#include <random>
+#include "dali/test/tensor_test_utils.h"
+#include "dali/test/test_tensors.h"
+#include "dali/kernels/transpose/transpose_test.h"
+#include "dali/kernels/scratch.h"
+
+namespace dali {
+namespace kernels {
+
+
+TEST(TransposeGPU, Test4DAll) {
+  std::mt19937_64 rng;
+  TensorListShape<> shape;
+  int N = 20;
+  int D = 4;
+
+  TestTensorList<int> in, out, ref;
+
+  TransposeGPU transpose;
+  ScratchpadAllocator sa;
+
+#ifdef NDEBUG
+  int max_extent = 70;
+#else
+  int max_extent = 30;
+#endif
+  std::uniform_int_distribution<int> shape_dist(1, max_extent);
+  std::uniform_int_distribution<int> small_shape_dist(1, 8);
+  std::bernoulli_distribution small_last_dim;
+
+  for (auto &perm : testing::Permutations4) {
```

Comment:
Would it use all transposition strategies available?

Reply:
It should.